src/jit/utils.cpp

```diff
@@ -1544,3 +1544,64 @@ unsigned            CountDigits(unsigned num, unsigned base /* = 10 */)
 }
 
 #endif // DEBUG
+
+
+double FloatingPointUtils::convertUInt64ToDouble(unsigned __int64 uIntVal) {
+    __int64 s64 = uIntVal; 
+    double d;
+    if (s64 < 0) {
+#if defined(_TARGET_XARCH_) && !defined(_MSC_VER)
```

Comment:
I think that this shodul just be #ifdef _TARGET_XARCH_

Because the JIT compiler on  XARCH is always doing this operation using a signed long:  
(double)(__int64)uIntVal;

The fact that VC gets also does it wrong is secondary and if VC changed  this (fixed it) we would still need to do the constant folding the same way as the RyuJIT does on XARCH.